CMakeLists.txt

```diff
@@ -81,11 +81,26 @@ endif(Python_EXECUTABLE AND NOT PYTHON_INSTDIR)
 ########################################################################
 # Checks what linear algebra packages are installed                    #
 ########################################################################
-set(MKL_THREAD_LAYER "Intel OpenMP" CACHE STRING "The thread layer to choose for MKL")
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+  set(MKL_THREAD_LAYER "Intel OpenMP" CACHE STRING "The thread layer to choose for MKL")
+else()
+  set(MKL_THREAD_LAYER "GNU OpenMP" CACHE STRING "The thread layer to choose for MKL")
+endif()
 find_package(MKL)
-if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
-  set_package_properties(MKL PROPERTIES TYPE OPTIONAL PURPOSE "Enables accelerated performance with MKL")
-endif (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+set_package_properties(MKL PROPERTIES TYPE OPTIONAL PURPOSE "Enables accelerated performance with MKL")
+
+#openmp with clang smaller version 10 and user defined reductions is buggy https://bugs.llvm.org/show_bug.cgi?id=44134
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 10.0)
+  # in case OpenMP was detected somewhere else
+  unset(OpenMP_FOUND)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Intel" AND MKL_FOUND)
+  # part of mkl
+  unset(OpenMP_FOUND)
```

Comment:
I think this should go

Reply:
because if we have gcc and mkl, we use gomp anyway. If we use MKL + icc we link aginst iopenmp which will be found again below